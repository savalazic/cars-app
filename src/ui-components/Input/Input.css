input {
  text-align: center;
  border: none;
  padding: 10px 30px;
  border-radius: 5px;
  box-shadow: 0 0 15px rgba(0, 0, 0, 0.1);
  outline: none;
  line-height: 1;
  color: #283048;
  width: 200px; }
  input[disabled] {
    opacity: 0.5;
    pointer-events: none; }
  input::placeholder {
    font-size: 14px;
    color: rgba(0, 0, 0, 0.3); }
