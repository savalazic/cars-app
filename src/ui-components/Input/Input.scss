@import '../../_variables';

$inputShadow: 0 0 15px rgba(black, 0.1);

input {
  text-align: center;
  border: none;
  padding: 10px 30px;
  border-radius: 5px;
  box-shadow: $inputShadow;
  outline: none;
  line-height: 1;
  color: $primaryColor;
  width: 200px;

  &[disabled] {
    opacity: 0.5;
    pointer-events: none;
  }

  &::placeholder {
    font-size: $fontSize;
    color: rgba(black, 0.3);
  }
}