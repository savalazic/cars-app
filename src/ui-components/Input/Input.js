// @flow
import React from 'react';

import './Input.css';

type Props = {
  onChange: () => void,
  placeholder: ?string,
  type: ?string,
  required: ?boolean,
};

const Input = ({
  onChange, placeholder, type, required,
}: Props) => (
  <input
    type={type}
    placeholder={placeholder}
    onChange={onChange}
    required={required}
  />
);

export default Input;
