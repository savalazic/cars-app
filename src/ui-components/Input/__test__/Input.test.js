import React from 'react';
import renderer from 'react-test-renderer';
import { shallow } from 'enzyme';

import Input from '../Input';

describe('Input component', () => {
  it('renders correctly', () => {
    const tree = renderer.create(<Input />).toJSON();
    expect(tree).toMatchSnapshot();
  });

  it('should render without throwing an error', () => {
    expect(shallow(<Input />).exists(<input />)).toBe(true);
  });
});
