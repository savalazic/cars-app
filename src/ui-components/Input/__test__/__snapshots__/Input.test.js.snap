// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Input component renders correctly 1`] = `
<input
  disabled={undefined}
  onChange={undefined}
  placeholder={undefined}
  required={undefined}
  type={undefined}
/>
`;
